### PR TITLE
[KEYCLOAK-16837] - Authz client still relying on refresh tokens when doing client credentials

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Comment by stianst** [comment]
Not issuing refresh tokens was a configuration switch wasn't it? So for current clients this would actually be a breaking change, which would result in a lot of sessions being created as the clients would get a refresh token, but not use them? At the very least this stuff needs to be documented in the migration guide

**Comment by pedroigor** [comment]
@stianst By default refresh tokens are disabled for client credentials. The changes here are about supporting this scenario which should be transparent to applications and users.

From a documentation perspective, this behavior is in the documentation where RTs are no longer issued by default.

**Review by sguilhen — DISMISSED** [review]

**Review by sguilhen — APPROVED** [review]

**Comment by pedroigor** [comment]
@sguilhen I've added tests. Also changed the `tokenResponse` to be stored after a `tryRefresh`. Sorry for not catching this before.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/AuthzClientCredentialsTest.java [hunk 1]
```diff
@@ -253,6 +255,32 @@ public void testSingleSessionPerUser() throws Exception {
         assertEquals(1, userSessions.size());
     }
 
+    @Test
+    public void testNoRefreshToken() throws Exception {
+        ClientsResource clients = getAdminClient().realm("authz-test-no-rt").clients();
+        AuthzClient authzClient = getAuthzClient("default-session-keycloak-no-rt.json");
+        org.keycloak.authorization.client.resource.AuthorizationResource authorization = authzClient.authorization();
+        AuthorizationResponse response = authorization.authorize();
```
Comment (mhajas) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/authz/AuthzClientCredentialsTest.java:263]: Does it make sense to check here whether response.getRefreshToken() is null?

**Review by mhajas — APPROVED** [review]
Thanks @pedroigor LGTM. I added one minor comment into the test. If you don't find it important feel free to merge it as is.

**Review by stianst — APPROVED** [review]

**Pull request merged**